File: oauth2/jwt/user-server/src/main/java/org/luvx/oauth2/res/controller/UserController.java
package org.luvx.oauth2.res.controller;

import java.time.LocalDateTime;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class UserController {
    @GetMapping("/hello")
    public String hello() {
        return "hello" + LocalDateTime.now();
    }

    @GetMapping("/admin/hello")
    public String admin() {
        return "admin";
    }
}

File: spring-boot-kotlin/src/main/kotlin/org/luvx/kotlin/controller/UserController.kt
package org.luvx.kotlin.controller

import org.luvx.java.service.UserReadService
import org.luvx.kotlin.entity.User
import org.luvx.kotlin.service.UserService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import javax.annotation.Resource

@RestController
@RequestMapping("/user")
class UserController {
    @Resource
    lateinit var userService: UserService

    @Resource
    lateinit var userReadService: UserReadService

    @GetMapping("/test")
    fun get() {
        val user = User()
        user.id = 50
        user.userName = "foo50"
        user.password = "bar50"
        user.age = 8
        userService.save(user)

        userReadService.select(user)
    }
}